core/eth/state.go

```diff
@@ -1,82 +1,275 @@
 package eth
 
 import (
+	"bytes"
 	"context"
+	"fmt"
 	"math/big"
+	"sync"
+	"sync/atomic"
 
 	"github.com/Layr-Labs/eigenda/common"
 	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	gcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 )
 
+// ChainState implements core.ChainState interface to query on-chain states
+// and filter socket updates through event logs to track the socket addresses
+// of operators in a map.
 type ChainState struct {
 	Client common.EthClient
-	Tx     core.Reader
+	Reader core.Reader
+	logger logging.Logger
+	// A cache map of the operator registry, key: operator id, value: socket string
+	SocketMap map[core.OperatorID]*string
+	// Mutex to access socket map
+	socketMu sync.Mutex
+	// The previous block number the socket map was updated at, inclusive
+	socketPrevBlockNumber atomic.Uint32
```

Comment:
Doesn't appear to need to be atomic any more, since access is fully contained within socketMu